documentation/TESTING_PLAN.md

```diff
@@ -16,8 +16,28 @@ In general, we're approaching testing bottom-up, with each unit function tested
 
 ### Unit-level tests
 
-- TBD 10
+- undo() should load the previous game state.
+- reset() should load the current level to its original state.
+- move(U) should allow the player to move up if it's a valid move.
+- move(D) should allow player to move down.
+- move(L) should allow player to move left.
+- move(R) should allow player to move right.
+- player should not be able to pass through impassable walls when move(U).
+- player should not be able to pass through impassable walls when move(D).
+- player should not be able to pass through impassable walls when move(L).
+- player should not be able to pass through impassable walls when move(R).
+- change_level(uint level_number) should allow user to switch levels.
+- moves() should return the number of moves made on the current board.
+- solved() should return true when all the boxes are on top of goal cells.
 
 ### Integration/E2E-level tests
 
-- TBD 10
+- When level is solved() congratulatory message should appear.
```

Comment:
Here, we aren't testing the back-end functions like `solved()` -- our test suite is pretending to be the user that isn't aware of technical details, just sees squares and UI. So, "should show congratulatory message when level is solved", "should display next level after clicking congraulatory message", "should move a player to the left when left arrow is pressed and destination square is empty", etc.